src/model/document.ts

```diff
@@ -48,6 +48,21 @@ export class ScannedDocument {
   get url() {
     return this.document.url;
   }
+
+  getNestedFeatures(): ScannedFeature[] {
```

Comment:
could take an optional argument which is a features array, then just push into the array as you recurse.